components/brave_rewards/browser/rewards_service_impl.cc

```diff
@@ -1367,6 +1424,154 @@ void RewardsServiceImpl::SetRewardsMainEnabledMigratedPref(bool enabled) {
   profile_->GetPrefs()->SetBoolean(prefs::kBraveRewardsEnabledMigrated, enabled);
 }
 
+void RewardsServiceImpl::SetCatalogIssuers(
+    std::unique_ptr<ads::IssuersInfo> info) {
+  if (!Connected()) {
+    return;
+  }
+
+  bat_ledger_->SetCatalogIssuers(info->ToJson());
+}
+
+void RewardsServiceImpl::AdSustained(
+    std::unique_ptr<ads::NotificationInfo> info) {
+  if (!Connected()) {
+    return;
+  }
+
+  bat_ledger_->AdSustained(info->ToJson());
+}
+
+void RewardsServiceImpl::URLRequest(const std::string& url,
+                                    const std::vector<std::string>& headers,
+                                    const std::string& content,
+                                    const std::string& content_type,
+                                    ledger::URL_METHOD method,
+                                    ledger::URLRequestCallback callback) {
+  net::URLFetcher::RequestType request_type = URLMethodToRequestType(method);
+
+  net::URLFetcher* fetcher = net::URLFetcher::Create(
+      GURL(url), request_type, this).release();
+  fetcher->SetRequestContext(g_browser_process->system_request_context());
+
+  for (size_t i = 0; i < headers.size(); i++)
+    fetcher->AddExtraRequestHeader(headers[i]);
+
+  if (!content.empty())
+    fetcher->SetUploadData(content_type, content);
+
+  fetchers_[fetcher] = callback;
+
+  fetcher->Start();
+}
+
+void RewardsServiceImpl::SaveConfirmationsState(const std::string& name,
```

Comment:
We don't need individual  methods for these. We just need a generic `Save` and `Load` method like AdsService which can be used by all state files